apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,51 @@
 'use strict';
 
+var $block = null;
+
+
 $(document).ready(function() {
   // Put app logic here
+  
+
+  $('[data-stack]').click(function () {
+    // $block = $(this).children().last().detach();
+      // The above variable and methods were moved to the else statement below
+    
+    if($block) {
+      var $lastBlock = $(this).children().last();
+      var $lastBlockValue = $lastBlock.data("block");
+      var $blockToBePlaced = $block.data("block");
+
+
+      if($blockToBePlaced < $lastBlockValue || !$lastBlockValue) {
+        //comparing the block that was clicked on to the last block in a stack
+        
+        $(this).append($block);
+        $block = null;
+        checkForWin();
+      }
+              
+    } else {
+      $block = $(this).children().last().detach();
```

Comment:
If you make `$lastBlock accessible to the scope of the whole function you can avoid duplicate logic here.

Ex:
`$block = $lastBlock.detach();`